main.go

```diff
@@ -111,6 +119,20 @@ func main() {
 		os.Exit(1)
 	}
 
+	if isCRPresent, err := r.IsCRPresent(); err != nil {
+		setupLog.Error(err, "problem checking if AWS SDK config exists")
+		os.Exit(1)
+	} else if isCRPresent {
+		setupLog.Info("AWS SDK config already exists. Skipping resource creation.")
+	} else {
+		setupLog.Info("creating AWS SDK config")
+		if err := r.CreateCR(); err != nil {
+			setupLog.Error(err, "unable to create AWS SDK config")
+			os.Exit(1)
+		}
+		setupLog.Info("AWS SDK config created successfully")
+	}
+
```

Comment:
Move above block to its own function, can club with the function that creates client. Or have sub-functions under createCRIfNeeded [CR does not exist, creation params specified]

Reply:
I moved the code to separate function. I am not sure if I got the sub-functions part.